### ISSUE TITLE
load_dataset with .zip files does not work

### ISSUE THREAD

**Issue by VinInn** [LAL/trackml-library#10]
It looks to me that
```
for event_id, hits, cells, particles, truth in load_dataset(file):
```
cannot work if file is the zip file train_sample.zip I downloaded from the kaggle site
as
```
zipinfo /Users/innocent/data/trackML_minitrain.zip | head
Archive:  /Users/innocent/data/trackML_minitrain.zip
Zip file size: 903185165 bytes, number of entries: 401
drwxrwxr-x  3.0 unx        0 bx stor 18-Apr-25 22:34 train_100_events/
-rw-rw-r--  3.0 unx   811007 tx defN 18-Apr-25 22:34 train_100_events/event000001054-particles.csv
-rw-rw-r--  3.0 unx 14012522 tx defN 18-Apr-25 22:34 train_100_events/event000001092-cells.csv
-rw-rw-r--  3.0 unx   824874 tx defN 18-Apr-25 22:34 train_100_events/event000001072-particles.csv
-rw-rw-r--  3.0 unx   557733 tx defN 18-Apr-25 22:34 train_100_events/event000001062-particles.csv
-rw-rw-r--  3.0 unx  4676360 tx defN 18-Apr-25 22:34 train_100_events/event000001093-hits.csv
-rw-rw-r--  3.0 unx  8506613 tx defN 18-Apr-25 22:34 train_100_events/event000001052-truth.csv
-rw-rw-r--  3.0 unx 12031041 tx defN 18-Apr-25 22:34 train_100_events/event000001013-cells.csv
```
while
https://github.com/LAL/trackml-library/blob/master/trackml/dataset.py#L133
request file name to match '^event...'

if I change to match '.*event...' i get
```
KeyError: "There is no item named 'event000001000-hits.csv' in the archive"
```

**Comment by dhrou** [comment]
HI Vincenzo,
Moritz should comment, but I am almost sure the zip file has to be unzipped.
What is supported is that individual .csv can be zipped, gzip etc…
David
-----------------------------------------------------------------------------------------------------
|     David Rousseau        rousseau@lal.in2p3.fr   skype:droussea                 |
|   LAL-Orsay, U Paris-Sud, CNRS/IN2P3, Université Paris-Saclay, France   |
|  snail:  LAL, CSO Orsay, Bat 200,  BP 34, 91898 Orsay Cedex.                   |
|  @ LAL     :  Bat 200 Pièce 142 : +33 (0)1 64 46 85 91                                 |
|  @ CERN :  40/1-D12                : +41 (22) 76 73857                                    |
-----------------------------------------------------------------------------------------------------

**Comment by msmk0** [comment]
This has been fixed in #9.
